File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
	int lcm = n * m;
	int q = n >= m ? n : m;
	int r = q != n ? n : m;

	while (true)
	{
		int temp = q % r;
		if (q % r == 0) break;
		q = r;
		r = temp;
	}
    answer.push_back(r);
    answer.push_back(lcm/r);
    
    return answer;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
using System;
public class Solution {
    
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        answer[0] = GCD(Math.Max(m, n), Math.Min(m, n));
        answer[1] = (n * m) / answer[0];
        return answer;
    }
    
    int GCD(int A,int B)
        {
            if (A == 0) return B;
            else return GCD(B % A, A);
        }
}